torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py

```diff
@@ -285,6 +332,56 @@ def annotate(self, model: torch.fx.GraphModule) -> torch.fx.GraphModule:
 
         return model
 
+    def _annotate_conv2d_bn(self, node: Node, operator_spec: Optional[OperatorSpec]) -> None:
+        """
+        Match the following pattern:
+
+          ... -> conv -> bn -> getitem[0] - ...
+
+        Annotate it to get the following pattern after prepare:
+
+                weight -> obs1
+                           |
+          ...  -> obs0 -> conv -> bn -> getitem[0] -> obs2 -> ...
+
+        Note: This is only used for QAT. In PTQ, batchnorm should already be fused into the conv.
+        """
+        if node.op != "call_function" or node.target != operator.getitem or node.args[1] != 0:
+            return
+        getitem_node = node
+        bn_node = getitem_node.args[0]
+        assert isinstance(bn_node, Node)
```

Comment:
What else can this be? If so, would it be better to return early instead of assert? Also instead of assert can we raise exception?

Reply:
This is mainly just for mypy. I think it's fine to assert here. We do the same for all the other annotations in this file too